PWGCF/Flow/Tasks/flowAnalysisGF.cxx

```diff
@@ -0,0 +1,686 @@
+    }
+  }
+
+  template <typename CollisionObject, typename TracksObject>
+  inline void FillEventQA(CollisionObject collision, TracksObject tracks)
+  {
+    registry.fill(HIST("globalTracks_centT0C"), collision.centFT0C(), tracks.size());
+    registry.fill(HIST("PVTracks_centT0C"), collision.centFT0C(), collision.multNTracksPV());
+    registry.fill(HIST("globalTracks_PVTracks"), collision.multNTracksPV(), tracks.size());
+    registry.fill(HIST("globalTracks_multT0A"), collision.multFT0A(), tracks.size());
+    registry.fill(HIST("globalTracks_multV0A"), collision.multFV0A(), tracks.size());
+    registry.fill(HIST("multV0A_multT0A"), collision.multFT0A(), collision.multFV0A());
+    registry.fill(HIST("multT0C_centT0C"), collision.centFT0C(), collision.multFT0C());
+    return;
+  }
+
+  Filter collisionFilter = nabs(aod::collision::posZ) < cfgVtxZ;
+  Filter trackFilter = nabs(aod::track::eta) < cfgEta && aod::track::pt > cfgPtmin&& aod::track::pt < cfgPtmax && ((requireGlobalTrackInFilter()) || (aod::track::isGlobalTrackSDD == (uint8_t) true)) && nabs(aod::track::dcaXY) < cfgDCAxy&& nabs(aod::track::dcaZ) < cfgDCAz;
+  using myTracks = soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::TrackSelection, aod::TracksDCA>>;
+
+  void processData(myCollisions::iterator const& collision, aod::BCsWithTimestamps const&, myTracks const& tracks)
+  {
+    registry.fill(HIST("hEventCount"), .5);
+
+    if (!collision.sel8())
+      return;
+
+    int Ntot = tracks.size();
+    if (Ntot < 1)
+      return;
+
+    const auto centrality = collision.centFT0C();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+
+    if (cfgUseAdditionalEventCut && !eventSelected(collision, tracks.size(), centrality))
+      return;
+    if (cfgFillQA)
+      FillEventQA(collision, tracks);
+    loadCorrections(bc.timestamp());
+    auto field = (cfgMagField == 99999) ? getMagneticField(bc.timestamp()) : cfgMagField;
+
+    processCollision<kReco>(collision, tracks, centrality, field);
+  }
+  PROCESS_SWITCH(flowAnalysisGF, processData, "Process analysis for non-derived data", true);
+
+  void processMCReco(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::Mults, aod::CentFT0Cs>>::iterator const& collision, aod::BCsWithTimestamps const&, soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::TrackSelection, aod::TracksDCA, aod::McTrackLabels>> const& tracks, aod::McParticles const&)
+  {
+    if (!collision.sel8())
+      return;
+    const auto centrality = collision.centFT0C();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    if (cfgUseAdditionalEventCut && !eventSelected(collision, tracks.size(), centrality))
+      return;
+    if (cfgFillQA)
+      FillEventQA(collision, tracks);
+    loadCorrections(bc.timestamp());
+    auto field = (cfgMagField == 99999) ? getMagneticField(bc.timestamp()) : cfgMagField;
+    processCollision<kReco>(collision, tracks, centrality, field);
```

Comment:
Why replicating this code?
I would recommend a single routine for processing reconstructed, be data or MC detector level, and then call it from both `process` functions